src/app/modules/form/public-form/public-form.middlewares.ts

```diff
@@ -0,0 +1,16 @@
+// TODO #4279: Remove after React rollout is complete
+import { StatusCodes } from 'http-status-codes'
+
+import { reactMigration } from '../../../config/config'
+import { ControllerHandler } from '../../core/core.types'
+
+export const injectFeedbackFormUrl: ControllerHandler = (req, res, next) => {
+  const formId = reactMigration.reactToAngularFeedbackFormId
+  if (!formId) {
+    return res
+      .status(StatusCodes.BAD_REQUEST)
+      .json({ message: 'No feedback form id provided' })
+  }
```

Comment:
I don't think this error represent a "Bad Request" 🤔 . If `config.reactMigration.reactToAngularFeedbackFormId` is not supplied, it's **our** mistake (not the caller), and should be a 500 unexpected error instead.

On that note don't we have a mechanism to ensure a value for the form is provided when the config object is instantiated?

Reply:
As discussed, the error that is thrown should not be a runtime error; rather the app shouldn't even work since this is a missing config. Hence I made the following changes
- remove the error handling from `public-form.middlewares`
- moved the `reactToAngularFeedbackFormId` env var from `IOptionalVarsSchema` to `ICompulsoryVarsSchema`
(sidenote: I've set `default` as `null`, because according to the [convict docs](https://github.com/mozilla/node-convict/tree/master/packages/convict) 'Every setting must have a default value'. However, ['you can set a default value to null and if your format doesn't accept null it will throw an error.'](https://github.com/mozilla/node-convict/tree/master/packages/convict#how-can-i-define-a-configuration-property-as-required-without-providing-a-default-value)

Tested the code on staging-alt2, it works! :) We get 502 Bad Gateway when `REACT_TO_ANGULAR_FEEDBACK_FORM_ID` is not supplied.

Reply:
> We get 502 Bad Gateway when REACT_TO_ANGULAR_FEEDBACK_FORM_ID is not supplied.

Wait, that was not my expectation 😰. I though having the config entry missing should cause the deployment to fail, but the service should stay up thanks to the old live instance(s) not being retired and not give 502s (maybe a 404 if a new route didn't get deployed?). Let's have another chat when you are in to discuss further 🙂

Reply:
Tested on Elastic Beanstalk, conclusion is we should use **rolling deployments** with >1 instance.
Upon deploying this, we get a "Failed to deploy application" error and the website is still accessible.